Create hexadecimal color syntax page (#14813)
mdn/content: files/en-us/web/css/color_value/hexadecimal_rgb/index.md

---
title: Hexadecimal color syntax
slug: Web/CSS/color_value/hexadecimal_rgb
tags:
  - Reference
  - color
browser-compat: css.types.color.rgb_hexadecimal_notation
---
{{CSSRef}}

> The **hexadecimal color syntax** is a notation allowing to describe a [sRGB](/en-US/docs/Glossary/RGB) color using its primary color components (red, green, blue) written as hexadecimal numbers, as well as its transparency. It can be used everywhere where a [`<color>`](/en-US/docs/Web/CSS/color_value) type is allowed.

Reviewer: ```suggestion
The **hexadecimal color syntax** is a notation for describing an [sRGB](/en-US/docs/Glossary/RGB) color using its primary color components (red, green, blue) written as hexadecimal numbers, as well as its transparency. It can be used everywhere a [`<color>`](/en-US/docs/Web/CSS/color_value) type is allowed.
```

## Syntax

```
#RGB        // The three-value syntax
#RGBA       // The four-value syntax
#RRGGBB     // The six-value syntax
#RRGGBBAA   // The eight-value syntax
```

> ## Value

Reviewer: Usually "Value" is an H3 under "Syntax".

Author: Fixed.

- `R` or `RR`

>   - : The _red_ component of the color, in case-insensitive hexadecimal number betwen `0` and `ff` (255)). If there is only one number, it is duplicated: `1` means `11`.

Reviewer: ```suggestion
  - : The _red_ component of the color, as a case-insensitive hexadecimal number between `0` and `ff` (255)). If there is only one number, it is duplicated: `1` means `11`.
```

- `G` or `GG`

>   - : The _green_ component of the color, in case-insensitive hexadecimal number betwen `0` and `ff` (255)). If there is only one number, it is duplicated: `c` means `cc`.

Reviewer: ```suggestion
  - : The _green_ component of the color, as a case-insensitive hexadecimal number between `0` and `ff` (255)). If there is only one number, it is duplicated: `c` means `cc`.
```

- `B` or `BB`

>   - : The _blue_ component of the color, in case-insensitive hexadecimal number betwen `0` and `ff` (255)). If there is only one number, it is duplicated: `9` means `99`.

Reviewer: ```suggestion
  - : The _blue_ component of the color, as a case-insensitive hexadecimal number between `0` and `ff` (255)). If there is only one number, it is duplicated: `9` means `99`.
```

- `A` or `AA` {{optional_inline}}

>   - : The _alpha_ component of the color, indicating its transparency, in case-insensitive hexadecimal number betwen `0` and `ff` (255)). If there is only one number, it is duplicated: `e` means `ee`. `0` represents a fully transparent color, and `ff`a fully opaque one.

Reviewer: ```suggestion
  - : The _alpha_ component of the color, indicating its transparency, as a case-insensitive hexadecimal number between `0` and `ff` (255)). If there is only one number, it is duplicated: `e` means `ee`. `0` represents a fully transparent color, and `ff`a fully opaque one.
```

> **Note:** The syntax is case-insensitive `#00ff00` is the same as `#00FF00`.

## Examples

```html hidden
/* Hexadecimal syntax for a fully opaque hot pink */
#f09 <div class="c1"></div>
#F09 <div class="c2"></div>
#ff0099 <div class="c3"></div>
#FF0099 <div class="c4"></div> 
```

> ```css hidden

Reviewer: Probably a matter of opinion, but I wouldn't hide these (especially not the CSS that sets the color). It's nice to be able to see the actual CSS that's doing the work.

I wouldn't hide the HTML either in this case, there's not much of it and I like being able to see all the code. But that's definitely more arguable.

Author: Fixed.

div {width: 40px; height: 40px;}
.c1 { background: #f09}
.c2 { background: #F09}
.c3 { background: #ff0099}
.c4 { background: #FF0099}
```

{{EmbedLiveSample("Examples", "100%", "450")}}

## Specifications

> {{Specifications}}

Reviewer: Will there be a spec_url in BCD for this?

Author: Yes, it landed in mdn/browser-compat-data#15762
